### PR TITLE
feat(odata-service-writer): mockserver logic seperation

### PR DESCRIPTION
**Pull request by martinjanus-sap**
Fix for #1793

- Clean up of odata-service-writer to use mock server generation logic from mockserver-config-writer.
- Change to `enhanceScripts` within mockserver-config-writer required as we dont to change existing mock-server script if it is added by another writer. 
- Remove ui5-mock.yaml from odata writer, no longer required
- Cleanup tests
- the `package.json` scrip re-ordering has changed due to the fs sequence of updates but the values are identical

### CODE REVIEW

File: packages/fiori-elements-writer/package.json [hunk 1]
```diff
@@ -31,9 +31,9 @@
         "!dist/**/*.map"
     ],
     "dependencies": {
+        "@sap-ux/fe-fpm-writer": "workspace:*",
```
Comment (longieirl) [inline packages/fiori-elements-writer/package.json:34]: Why is the order changed?

File: packages/odata-service-writer/src/index.ts [hunk 2]
```diff
@@ -133,11 +108,10 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
     if (service.metadata) {
         // copy existing `ui5.yaml` as starting point for ui5-mock.yaml
         if (paths.ui5Yaml && ui5Config) {
-            const ui5MockConfig = await UI5Config.newInstance(ui5Config.toString());
-            ui5MockConfig.addMockServerMiddleware(service.path, await generateODataAnnotationConfig(basePath, fs));
-            fs.write(join(dirname(paths.ui5Yaml), 'ui5-mock.yaml'), ui5MockConfig.toString());
-
-            // also add mockserver middleware to ui5-local.yaml
+            const webappPath = await getWebappPath(basePath);
```
Comment (longieirl) [inline packages/odata-service-writer/src/index.ts:111]: The flow of execution here is causing the changes to the package.json's, if you move this code after the ui5LocalConfig.addMockServerMiddleware, you should see less file differences.

File: packages/mockserver-config-writer/src/mockserver-config/package-json.ts [hunk 3]
```diff
@@ -58,7 +58,9 @@ function enhanceDependencies(
 function enhanceScripts(fs: Editor, packageJson: Package): void {
     packageJson.scripts ||= {};
     packageJson.scripts['start-mock'] =
-        copyStartScript(packageJson.scripts.start) || `fiori run --config ./ui5-mock.yaml --open \"/\"`;
+        copyStartScript(packageJson.scripts.start) ??
+        packageJson.scripts['start-mock'] ??
```
Comment (longieirl) [inline packages/mockserver-config-writer/src/mockserver-config/package-json.ts:62]: This change ensures that if there is an existing mock-server added by another writer, we dont touch it.

File: packages/odata-service-writer/src/index.ts [hunk 4]
```diff
@@ -133,11 +108,13 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
     if (service.metadata) {
         // copy existing `ui5.yaml` as starting point for ui5-mock.yaml
         if (paths.ui5Yaml && ui5Config) {
-            const ui5MockConfig = await UI5Config.newInstance(ui5Config.toString());
-            ui5MockConfig.addMockServerMiddleware(service.path, await generateODataAnnotationConfig(basePath, fs));
-            fs.write(join(dirname(paths.ui5Yaml), 'ui5-mock.yaml'), ui5MockConfig.toString());
```
Comment (longieirl) [inline packages/odata-service-writer/src/index.ts:138]: If there are no more references to ui5-mock, please remove it.

File: packages/odata-service-writer/test/unit/index.test.ts [hunk 5]
```diff
@@ -98,13 +98,12 @@ describe('generate', () => {
         });
 
         it('Nested folder structure - all files updated', async () => {
-            const packagePath = join(testDir, 'package.json');
+            const packagePath = join(root, 'package.json');
```
Comment (longieirl) [inline packages/odata-service-writer/test/unit/index.test.ts:101]: Why was this changed?

**Review by Klaus-Keller — APPROVED** [review]
Thanks @martinjanus-sap!
- I reviewed mockserver-config-writer and code changes look good
- Did not run a manual test. just visual review
- Text coverage is great

Approved from my side.

**Review by devinea — APPROVED** [review]
Code changes are clear. 
Snapshot updates are expected. 
changeset 👍

File: packages/odata-service-writer/src/index.ts [hunk 6]
```diff
@@ -133,11 +108,13 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
...
-
-            // also add mockserver middleware to ui5-local.yaml
+            const webappPath = await getWebappPath(basePath);
```
Comment (devinea) [inline packages/odata-service-writer/src/index.ts:111]: `const webappPath = await getWebappPath(basePath, fs);`
@martinjanus-sap I think the `fs` param should be passed here to support memfs

Comment (longieirl) [reply packages/odata-service-writer/src/index.ts:111]: Thanks @devinea for the feedback. @martinjanus-sap and I will work to get this changed.
